frameworks/js-bindings/bindings/manual/network/XMLHTTPRequest.cpp

```diff
@@ -724,10 +731,28 @@ JS_BINDED_FUNC_IMPL(MinXmlHttpRequest, send)
     _setHttpRequestHeader();
     _sendRequest(cx);
     _notify(_onloadstartCallback);
+    
+    //begin schedule for timeout
+    if(_timeout > 0)
+    {
+        _scheduler->scheduleUpdate(this, 0, false);
```

Comment:
Ok, I think we should do the following: 
1. Call HttpClient:: setTimeoutForConnect when setting the _timeout value
2. When we receive response, we can check the response code, if it equals to 408, it's a timeout
3. Notify the timeout callback